src/exometer_report_influxdb.erl

```diff
@@ -264,3 +274,48 @@ evaluate_timestamp_opt(Term) when is_boolean(Term) ->
     {Term, ?DEFAULT_PRECISION};
 evaluate_timestamp_opt(_) ->
     exit(invalid_timestamp_option).
+
+-spec del_indices(list(), [integer()]) -> list().
+del_indices(List, Indices) ->
+    SortedIndices = lists:reverse(lists:usort(Indices)),
+    case length(SortedIndices) == length(Indices) of
+        true -> del_indices1(List, SortedIndices);
+        false -> exit({invalid_indices, Indices})
+    end.
+
+-spec del_indices1(list(), [integer()]) -> list().
+del_indices1(List, []) -> List;
+del_indices1([], Indices = [ _Index | _Indices1 ]) -> exit({too_many_indices, Indices});
+del_indices1(List, [Index | Indices]) when length(List) >= Index, Index > 0 ->
+    {L1, [_|L2]} = lists:split(Index-1, List),
+    del_indices1(L1 ++ L2, Indices);
+del_indices1(_List, Indices) ->
+    exit({invalid_indices, Indices}).
+
+-spec evaluate_subscription_tags(list(), [{atom(), value()}]) -> {list(), map()}.
+evaluate_subscription_tags(Metric, Tags) ->
+    evaluate_subscription_tags(Metric, Tags, [], []).
+
+-spec evaluate_subscription_tags(list(), [{atom(), value()}], [{atom(), value()}],
+                                 [integer()]) -> {list(), map()}.
+evaluate_subscription_tags(Metric, [], TagAkk, PosAkk) ->
+    MetricName = del_indices(Metric, PosAkk),
+    {MetricName, maps:from_list(TagAkk)};
+evaluate_subscription_tags(Metric, [{Key, {from_name, Pos}} | TagOpts], TagAkk, PosAkk)
+    when is_number(Pos), length(Metric) >= Pos, Pos > 0 ->
+    NewTagAkk = TagAkk ++ [{Key, lists:nth(Pos, Metric)}],
+    NewPosAkk = PosAkk ++ [Pos],
+    evaluate_subscription_tags(Metric, TagOpts, NewTagAkk, NewPosAkk);
+evaluate_subscription_tags(Metric, [TagOpt = {Key, {from_name, Name}} | TagOpts],
```

Comment:
This way you can alter the metric id using keywords (and not only with positional parameters).